client/control/tests/reducers/test_columnReducer.js

```diff
@@ -0,0 +1,129 @@
+import appReducer from 'control/reducers';
+import * as actions from 'control/actions/ColumnActions';
+
+import {
+  initialState,
+} from 'control/tests/fixtures';
+
+/**
+ * Utility function to clone the initialState column array and its values
+ * (This prevents nested objects from being cloned by reference)
+ *
+ * @return {Array} New array with cloned initialState values
+ */
+const cloneInitialArray = () => JSON.parse(JSON.stringify(initialState.columns));
+
+
+/**
+ * Utility function to get the index of a column
+ * which is NOT shown by default.
+ *
+ * @return {Number} Index of non-default column to test against.
+ */
+const getNondefaultColumn = () => {
+  let foundIndex;
+  initialState.columns
+    .find((col, index) => {
+      foundIndex = index;
+      return !col.enabled;
+    });
+
+  if (!foundIndex) {
+    console.warn('Warning! No non-default columns to test against.');
+  }
+  return foundIndex;
+};
+
+/**
+ * Column reducer tests
+ */
+describe('Column reducer', () => {
+  it('should return initial state by default', () => {
+    expect(appReducer(undefined, {})).toEqual(initialState);
+  });
+
+  it('should have default columns to display', () => {
+    const defaults = [
+      'name',
+      'action',
+      'enabled',
+      'last_updated',
+      'metadata',
+    ];
+
+    // get the columns that are enabled,
+    // and their shorthand values
+    const foundDefaults = cloneInitialArray()
+      .filter(col => col.enabled)
+      .map(col => col.value);
+
+    // arrays of default enableds should match
+    expect(foundDefaults).toEqual(defaults);
+  });
+
+  it('should handle UPDATE_COLUMN (enable)', () => {
+    const expectedColumns = cloneInitialArray();
+
+    // grab a non-default'd column
+    const colId = getNondefaultColumn();
+    expectedColumns[colId].enabled = true;
+
+    expect(appReducer(undefined, {
+      type: actions.UPDATE_COLUMN,
+      index: colId,
+      isActive: true,
+    })).toEqual({
+      ...initialState,
+      columns: expectedColumns,
+    });
+  });
+
+  it('should handle UPDATE_COLUMN (disable)', () => {
+    const expectedColumns = cloneInitialArray();
+    expectedColumns[0].enabled = false;
```

Comment:
This assumes the first column will always be a default column. Is that a safe assumption?